pkg/api/v1/conversion.go

```diff
@@ -625,10 +622,7 @@ func Convert_v1_Pod_To_api_Pod(in *Pod, out *api.Pod, s conversion.Scope) error
 		// back to the caller.
 		in.Spec.InitContainers = values
 		// Call defaulters explicitly until annotations are removed
-		for i := range in.Spec.InitContainers {
-			c := &in.Spec.InitContainers[i]
-			SetDefaults_Container(c)
-		}
+		SetDefaults_InitContainers(in.Spec.InitContainers)
```

Comment:
is there a reason we can't just do this?
```
tmpPod := &Pod{Spec:PodSpec{InitContainers:values}}
SetObjectDefaults_Pod(tmpPod)
in.Spec.InitContainers = tmpPod.Spec.InitContainers
```

the copied SetDefaults_InitContainers is going to drift